Update "GDScript reference" (operators and `match`) (#7323)
godotengine/godot-docs: tutorials/scripting/gdscript/gdscript_basics.rst

Also, using ``\`` followed by a newline inside a string will allow you to continue it in the next line, without
inserting a newline character in the string itself.

> GDScript also supports :ref:`Format Strings <doc_gdscript_printf>`.

Reviewer: "Format strings" isn't a proper word per se, so I wouldn't capitalize it:

```suggestion
GDScript also supports :ref:`format strings <doc_gdscript_printf>`.
```

        [pattern](s):
            [block]

.. warning::

>     ``match`` is more type strict than the ``==`` operator. For example ``1`` will NOT match ``1.0``. The only exception is ``String`` vs ``StringName`` matching.

Reviewer: ```suggestion
    ``match`` is more type-strict than the ``==`` operator. For example, ``1`` will **not** match ``1.0``. The only exception is ``String`` vs ``StringName`` matching.
```

3. Remove any ``break``\ s.
4. Change ``default`` to a single underscore.

.. note::

>     The special ``continue`` behavior in ``match`` supported in 3.x was removed prior to the 4.0 release.

Reviewer: Make the wording more explicit:

```suggestion
    The special ``continue`` behavior in ``match`` supported in 3.x was removed in Godot 4.0.
```

        label.text += character_name + " took " + str(damage) + " damage."

Awaiting for signals or coroutines
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The ``await`` keyword can be used to create `coroutines <https://en.wikipedia.org/wiki/Coroutine>`_

> which waits until a signal is emitted before continuing execution. Using the ``await`` keyword with a signal or a

Reviewer 2: ```suggestion
which wait until a signal is emitted before continuing execution. Using the ``await`` keyword with a signal or a
```
